### DISCUSSION TITLE
OpenUTAU Idea's

### DISCUSSION THREAD

**Discussion by EnjouAkuma** [stakira/OpenUtau#441] [Ideas]
**First time posting anything on Github, and I am no too familiar with it, so please tell me if I had said this in the wrong place. I was hoping that maybe stakira or one of the developer's would see this and maybe think of using these ideas I had? Some of these might already be planned, but I am not in the OpenUTAU server, so I wouldn't know. Anyway, here are a list of ideas I have for OpenUTAU!** 

Ya'll this is old, please ignore or delete this post XD

**Comment by nagotown** [comment]
1. UTAU voicebanks cannot have AI capabilities, they are concatenative only. AI voicebanks require actual singing data. you can make an NNSVS voicebank out of rendered UTAU singing though.

2. Custom themes are already planned but not a priority as it is purely cosmetic and doesn't do anything for functionality. If you know how to compile from source and want to do it every update then it would be plausible for yourself.

3. Auto pitch tuning will never be as good as tuning manually, generally it's for quick things or as a base. OpenUtau would need AI models to reference for more realistic pitch and there's already EnuPitch for that although it doesn't work right now.

4. The availability of a growl parameter depends on the developer of the resampler you use. Moresampler has a growl flag (MG) and there's a patcher for other resamplers to add a growl flag.
The classic renderer (which uses UTAU resamplers) cannot handle some parameters due to the way they are built. worldline-r is a version of the worldline resampler that supports extra parameters due to being built into OpenUtau rather than an exe file that connects to the software. Being fair to other resamplers in terms of parameter support is not something that will happen because the majority of resamplers are independently developed and the devs don't implement everything.

5. Kanji to romaji does not make any sense because there are zero uses of kanji in usts other than to denote an append. Kanji have different pronunciations most of the time so it would be more work than it's worth.

6. OpenUtau is not for speech but you can make a ust manually for it. TalQu is not in any way affiliated with UTAU and is for AI voicebanks. TalQu is free, you were just looking at the donation button.

7. The majority of phonemizers are developed by the community so you'll have to request them. CV English is not very common and is hard to work with so you would need to get someone to agree to it.

8. You can use utaformatix for converting usts to vsqx and vice versa.

9. Same as above, with the addition that acep files will probably not be added to utaformatix because one of the people who contribute to it works on ACE and their company said they can't.

10. This is being actively worked on and you can try a preview of it on another branch of the repository if I recall correctly.

**Comment by kokoropaint** [comment]
Just wanted to add to this that you can use [this program](https://ameblo.jp/maiko3utau/entry-12660767498.html)  to prepare Japanese lyrics for synth input, it also lets you specify alternate readings for kanji/etc before conversion. Considering that OU has batch lyric editing and the points that nago mentioned it makes a lot more sense for this to remain an external tool.

I know paid resources aren't accessible to you but to be honest, if you want AI auto tuning I would just buy a recent synthV ai vocal and use utaformatix to convert to USTX, lol. it's not something this project is aiming for and not something that many people would use. The pitch conversion that utaformatix does works very well as it is.

**Reply by shuntia** [reply]
synthV is partially free, so it is now an actual viable option.

**Comment by oxygen-dioxide** [comment]
> **First time posting anything on Github, and I am no too familiar with it, so please tell me if I had said this in the wrong place. I was hoping that maybe stakira or one of the developer's would see this and maybe think of using these ideas I had? Some of these might already be planned, but I am not in the OpenUTAU server, so I wouldn't know. Anyway, here are a list of ideas I have for OpenUTAU!**

@EnjouAkuma Diffsinger already supports converting classical voicebanks into AI voicebanks. It will be smoother than classical resamplers. [Here is a demo](https://www.bilibili.com/video/av651966265)

**Reply by EnjouAkuma** [reply]
I made this well before I knew about DIffsinger and the colab notebook is down atm, so I can't use that.
